.github/workflows/e2e-pw-pull-request-only-changed.yml

```diff
@@ -0,0 +1,75 @@
+name: E2E Playwright Tests - Pull Request - Only Changed
+
+on:
+    pull_request:
+        branches:
+            - develop
+            - trunk
+    workflow_dispatch:
+    workflow_call:
+        inputs:
+            wcpay-use-build-artifact:
+                type: boolean
+                required: false
+                default: false
+            repo-branch:
+                type: string
+                required: false
+                description: 'Branch to be used for running tests'
+
+env:
+    E2E_GH_TOKEN: ${{ secrets.E2E_GH_TOKEN }}
+    WCP_DEV_TOOLS_REPO: ${{ secrets.WCP_DEV_TOOLS_REPO }}
+    WCP_DEV_TOOLS_BRANCH: 'trunk'
+    TRANSACT_PLATFORM_SERVER_REPO: ${{ secrets.TRANSACT_PLATFORM_SERVER_REPO }}
+    WC_SUBSCRIPTIONS_REPO: ${{ secrets.WC_SUBSCRIPTIONS_REPO }}
+    E2E_BLOG_ID: ${{ secrets.E2E_BLOG_ID }}
+    E2E_BLOG_TOKEN: ${{ secrets.E2E_BLOG_TOKEN }}
+    E2E_USER_TOKEN: ${{ secrets.E2E_USER_TOKEN }}
+    WC_E2E_SCREENSHOTS: 1
+    E2E_SLACK_CHANNEL: ${{ secrets.E2E_SLACK_CHANNEL }}
+    E2E_SLACK_TOKEN: ${{ secrets.E2E_SLACK_TOKEN }}
+    E2E_USE_LOCAL_SERVER: false
+    E2E_RESULT_FILEPATH: 'tests/e2e/results.json'
+    WCPAY_USE_BUILD_ARTIFACT: ${{ inputs.wcpay-use-build-artifact }}
+    WCPAY_ARTIFACT_DIRECTORY: 'zipfile'
+    NODE_ENV: 'test'
+    FORCE_E2E_DEPS_SETUP: true
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    wcpay-e2e-tests-only-changed:
+        runs-on: ubuntu-latest
+        if: github.event.pull_request.draft == true
+        name: WC - latest
+
+        env:
+            E2E_WP_VERSION: 'latest'
+            E2E_WC_VERSION: 'latest'
```

Comment:
The `E2E tests - All` workflow runs tests against different versions of WP and WC (7.7.0, latest and beta, for example). Would we want to consider this in this workflow? There will be times when tests will pass against latest but fail against beta, like with the **Coming Soon mode** introduced in the latest WC release.